src/utils/collapseSourcemaps.ts

```diff
@@ -33,14 +30,14 @@ interface SourceMapSegmentObject {
 }
 
 class Link {
-	mappings: SourceMapSegmentVector[][];
+	mappings: SourceMapMappings;
 	names: string[];
-	sources: Source[];
+	sources: (Source | Link)[];
 
-	constructor(map: { mappings: SourceMapSegmentVector[][]; names: string[] }, sources: Source[]) {
+	constructor(map: { mappings: number[][][]; names: string[] }, sources: (Source | Link)[]) {
 		this.sources = sources;
 		this.names = map.names;
-		this.mappings = map.mappings;
+		this.mappings = map.mappings as SourceMapMappings;
```

Comment:
I believe this type casts and many of the other remaining type casts can be avoided if instead of using `SourceMapMappings`, `SourceMapLine` and `SourceMapSegment` we internally use `number[][][]`, `number[][]` and `number[]. To make this clearer, we could introduce our own types to encapsulate these. This would get rid of the type casts here and in lines 61, 96, and 128, and would require using our own types in lines 33, 51, and 82.

Reply:
That would be an even better solution. Only point is that I would like to avoid an external dependency from types.d.ts to sourcemap-codec as this would introduce a completely new external dependency onto the published bundle just for its types. Instead, I would prefer to just copy the types, as few as they are. Improving the types of magic-string would definitely be a good idea, also considering both sourcemap-codec and magic-string are originally from the same author. Until this is done, your suggestion is a good approach.

Reply:
Done.